pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -620,30 +738,43 @@ func (plugin *kubenetNetworkPlugin) syncEbtablesDedupRules(macAddr net.HardwareA
 		return
 	}
 
-	klog.V(3).Infof("Filtering packets with ebtables on mac address: %v, gateway: %v, pod CIDR: %v", macAddr.String(), plugin.gateway.String(), plugin.podCidr)
+	// ensure custom chain exists
 	_, err = plugin.ebtables.EnsureChain(utilebtables.TableFilter, dedupChain)
 	if err != nil {
 		klog.Errorf("Failed to ensure %v chain %v", utilebtables.TableFilter, dedupChain)
 		return
 	}
 
+	// jump to custom chain to the chain from core tables
 	_, err = plugin.ebtables.EnsureRule(utilebtables.Append, utilebtables.TableFilter, utilebtables.ChainOutput, "-j", string(dedupChain))
 	if err != nil {
 		klog.Errorf("Failed to ensure %v chain %v jump to %v chain: %v", utilebtables.TableFilter, utilebtables.ChainOutput, dedupChain, err)
 		return
 	}
 
-	commonArgs := []string{"-p", "IPv4", "-s", macAddr.String(), "-o", "veth+"}
-	_, err = plugin.ebtables.EnsureRule(utilebtables.Prepend, utilebtables.TableFilter, dedupChain, append(commonArgs, "--ip-src", plugin.gateway.String(), "-j", "ACCEPT")...)
-	if err != nil {
-		klog.Errorf("Failed to ensure packets from cbr0 gateway to be accepted")
-		return
+	// per gateway rule
+	for idx, gw := range plugin.podGateways {
```

Comment:
@khenidak podGateways needs to be accessed under the plugin lock. This function is called from setup() which is called from SetUpPod() and neither of those grab the lock.  Same for the access of podCIDRs here.

Reply:
Same as PodCIDR, it is read only across the entire thing.  And no read attempt will happen until Status() call returns success, which does not do that until happen until Event() sets it. There is no point of having a lock there.

Reply:
@khenidak Event() may be called at any time, in reality it will be called whenever kubelet.go's fastStatusUpdateOnce() sees that the node's PodCIDR has been updated. It's not only called at kubelet startup. Thus I believe there may be concurrent access if Event() is called at the same time as setup() is called when hairpin mode is PromiscuousBridge. Might also be some for plugin.netConfig too.

Reply:
Nevermind, you're right that Event() must be called before Status() returns OK and allows pods to be set up.